core/src/components/button/button.ionic.scss

```diff
@@ -131,15 +131,23 @@
 ::slotted(ion-icon[slot="start"]),
 ::slotted(ion-icon[slot="end"]),
 ::slotted(ion-icon[slot="icon-only"]) {
-  font-size: globals.$ion-font-size-400;
+  font-size: globals.$ion-font-size-500;
 }
 
 :host(.button-small),
+:host(.button-medium) {
+  ::slotted(ion-icon[slot="start"]),
+  ::slotted(ion-icon[slot="end"]),
+  ::slotted(ion-icon[slot="icon-only"]) {
+    font-size: globals.$ion-font-size-500;
+  }
+}
```

Comment:
Why are these needed when the [default](https://github.com/ionic-team/ionic-framework/pull/30414/files?file-filters%5B%5D=.html&file-filters%5B%5D=.scss&file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx&show-viewed-files=true#diff-e7130eac638157a72b40aacb382be024fcd67337af1ab1d88b2c5a5f2f28912dR134) (line 134) is already 500?